spec/lib/primo/search_spec.rb

```diff
@@ -213,4 +213,33 @@
       }
     end
   end
+
+  context "timetout retry" do
+    before do
+      Primo.configure do |config|
+        config.enable_retries = true
+      end
+    end
+
+    let(:options) {
+      q = Primo::Search::Query.new(
+        precision: :contains,
+        field: :title,
+        value: "otter",
+        operator: :OR,
+      )
+      { q: }
+    }
+
+    it "should retry if a timeout occurs" do
+      allow(Primo.configuration).to receive(:timeout).and_raise(Net::ReadTimeout).once
+      # [TODO] Fix potential false-positive when checking an exception DOES NOT occur
+      expect { Primo::Search::get(options) }.not_to raise_error("Primo timeout retry limit exceeded")
```

Comment:
I'm not clear on what is the false positive you expect to happen?

Reply:
@dkinzer This is a warning from rspec.  It is possible that an error could occur that is not specifically Net::ReadTimeout.  I tried using a generic execption, but that test always fails, because we are, infact, raising exceptions.  OTOH, I tried disabling the exception, per the recommendation from rspec, but that has to be done in such a way as to disable false-positive checking for all specs.